libbpf-rs/tests/test.rs

```diff
@@ -242,3 +246,89 @@ fn test_object_link_pin() {
     link.unpin().expect("failed to unpin prog");
     assert!(!Path::new(path).exists());
 }
+
+#[test]
+fn test_ringbuf() {
+    bump_rlimit_mlock();
+
+    let mut obj = get_test_object("ringbuf.bpf.o");
+    let prog = obj
+        .prog("handle__sys_enter_getpid")
+        .expect("error finding program")
+        .expect("failed to find program");
+    let _link = prog.attach().expect("failed to attach prog");
+
+    static mut v1: i32 = 0;
+    static mut v2: i32 = 0;
+
+    fn callback1(data: &[u8]) -> i32 {
+        let mut value: i32 = 0;
+        plain::copy_from_bytes(&mut value, data).expect("Wrong size");
+
+        unsafe {
+            v1 = value;
+        }
+
+        0
+    }
+
+    fn callback2(data: &[u8]) -> i32 {
+        let mut value: i32 = 0;
+        plain::copy_from_bytes(&mut value, data).expect("Wrong size");
+
+        unsafe {
+            v2 = value;
+        }
+
+        0
+    }
+
+    let mut mgr = libbpf_rs::RingBufferManager::default();
+
+    mgr.poll(Duration::from_millis(100))
+        .expect_err("This shouldn't work until we add a ringbuf");
+
+    mgr.consume()
+        .expect_err("This shouldn't work until we add a ringbuf");
+
+    let map1 = obj
+        .map_immut("ringbuf1")
+        .expect("Error getting ringbuf1 map")
+        .expect("Failed to get ringbuf1 map");
+
+    mgr.add_ringbuf(map1, callback1 as fn(&[u8]) -> i32)
```

Comment:
Because functions in Rust always have unique types (regardless of signature), the cast is necessary to coerce the compiler to use a function pointer instead. The compiler isn't smart enough to figure out that it should be using a function pointer, so we need to do a cast here. See https://stackoverflow.com/questions/27895946/expected-fn-item-found-a-different-fn-item-when-working-with-function-pointer

Reply:
If you can think of a better way to deal with this, I'm happy to discuss. If not, I'll update the documentation to explicitly mention this.